docworks-dts/lib/dts-generator.js

```diff
@@ -49,6 +54,61 @@ function getDtsType(type, options = {}) {
   throw new Error(`Unable to convert type ${type} to valid dts type`)
 }
 
+// Docs Type Definition: https://github.com/wix-private/p13n/blob/master/wixplorer/wix-docs-server/proto/docs/README.md
+function isNewDocsTypesDefinition(type) {
+  return type.some(currentType => currentType.nativeType || currentType.complexType || currentType.referenceType)
+}
+
+function getDtsFromDocsType(docsTypes) {
+  if(isUnionType(docsTypes)){
+    return docsTypes.map(t => getDtsFromDocsType([t])).join(' | ')
+  }
+  else{
+    const docsType = docsTypes[0]
+    if(docsType.nativeType || docsType.referenceType){
```

Comment:
it's strange to have a recursive that doesn't start with the stop condition.
imo the function should start with `if(docsType.nativeType || docsType.referenceType)` condition and all other recursive calls should be in the else block

Reply:
i prefer starting with stop condition, but it's not critical